File: 프로그래머스/2/42586. 기능개발/기능개발.swift
import Foundation

func solution(_ progresses:[Int], _ speeds:[Int]) -> [Int] {
    
    let N = progresses.count
    
    var progresses = progresses
    
    var result: [Int] = []
    var deploy: Int = 0
    
    while deploy != N {
        
        var curDeploy = 0
        
        for i in 0..<N {
            
           guard progresses[i] != -1 else { continue }
            
            if progresses[i] < 100 {
                progresses[i] = min(100, progresses[i] + speeds[i])
            }
        
            if progresses[i] >= 100 && deploy + curDeploy >= i {
                curDeploy += 1
                progresses[i] = -1
            }
        }
        
        if curDeploy > 0 {
            result.append(curDeploy)
            deploy += curDeploy
        }
    }
    
    return result
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    
    n = len(progresses)
    
    answer = []
    pop_idx = 0
    
    while True:
        if pop_idx == n: break
        
        # 미완료 작업 진도 수행
        for i in range(pop_idx, n):
            progresses[i] += speeds[i]
        
        # 완료 작업 수 카운트
        finish = 0
        while pop_idx < n and progresses[pop_idx] >= 100:
            pop_idx += 1
            finish += 1
        
        # 완료 작업 존재 시 카운트를 배열에 추가 
        if finish > 0:
            answer.append(finish)
    
    return answer